src/library/Cardano/CoinSelection/Fee.hs

```diff
@@ -131,35 +131,73 @@ data FeeOptions i o = FeeOptions
     } deriving Generic
 
 newtype FeeAdjustmentError
-    = ErrCannotCoverFee Fee
-    -- ^ UTxO exhausted during fee covering
-    -- We record what amount missed to cover the fee
+    = CannotCoverFee Fee
+    -- ^ Indicates that the given map of additional inputs was exhausted while
+    --   attempting to select extra inputs to cover the required fee.
+    --
+    -- Records the shortfall between the required fee and the total value
+    -- of currently selected inputs.
+    --
     deriving (Show, Eq)
 
--- | Given the coin selection result from a policy run, adjust the outputs for
--- fees, potentially returning additional inputs that we need to cover all
--- fees.
+-- | Adjusts the given 'CoinSelection' in order to pay for a __transaction__
+--   __fee__, required in order to publish the selection as a transaction on
+--   a blockchain.
+--
+-- == Background
+--
+-- Implementations of 'Cardano.CoinSelection.CoinSelectionAlgorithm' generally
+-- produce coin selections that are /exactly balanced/, satisfying the
+-- following equality:
+--
+-- >>> sumInputs c = sumOutputs c + sumChange c
+--
+-- In order to pay for a transaction fee, the above equality must be
+-- transformed into an /inequality/:
+--
+-- >>> sumInputs c > sumOutputs c + sumChange c
+--
+-- The difference between these two sides represents value to be paid /by the/
+-- /originator/ of the transaction, in the form of a fee:
+--
+-- >>> sumInputs c = sumOutputs c + sumChange c + fee
+--
+-- == The Adjustment Process
+--
+-- In order to generate a fee that is acceptable to the network, this function
+-- adjusts the 'change' and 'inputs' of the given 'CoinSelection', consulting
+-- the 'FeeEstimator' as a guide for how much the current selection would cost
+-- to publish as a transaction on the network.
+--
+-- == Methods of Adjustment
+--
+-- There are two methods of adjustment possible:
+--
+--  1. The __'change'__ set can be /reduced/, either by:
+--
+--      a. completely removing a change value from the set; or by
+--
+--      b. reducing a change value to a lower value.
+--
+--  2. The __'inputs'__ set can be /augmented/, by selecting additional inputs
+--     from the specified 'CoinMap' argument.
+--
+-- == Dealing with Dust Values
+--
+-- If, at any point, a change value is generated that is less than or equal
+-- to the 'DustThreshold', this function will eliminate that change value
+-- from the 'change' set, redistributing the eliminated value over the remaining
+-- change values, ensuring that the total value of all 'change' is preserved.
```

Comment:
Perhaps it would be worth referencing the 'coalesceDust' function here, as the documentation comment for that function provides a good description of how dust elimination works.

Reply:
> Perhaps it would be worth referencing the `coalesceDust` function here, as the documentation comment for that function provides a good description of how dust elimination works.

Fixed in 8108efb684a937d3dfe3d855b3d327f7a439d466.